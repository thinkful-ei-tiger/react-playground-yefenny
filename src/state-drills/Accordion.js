import React from 'react';

class Accordion extends React.Component {
  static defaultProps = {
    sections: []
  };
  constructor(props) {
    super(props);
    this.state = {
      section: null
    };
  }
  handleClick = (index) => {
    this.setState({
      section: index
    });
  };
  render() {
    return (
      <ul>
        {this.props.sections.map((section, index) => {
          return (
            <li key={index}>
              <button onClick={() => this.handleClick(index)}>
                {section.title}
              </button>
              {this.state.section === index && <p>{section.content}</p>}
            </li>
          );
        })}
      </ul>
    );
  }
}

export default Accordion;
