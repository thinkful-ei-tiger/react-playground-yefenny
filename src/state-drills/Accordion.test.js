import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordion from './Accordion';

describe('Accordion component', () => {
  const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.'
    },
    {
      title: 'Section 2',
      content:
        'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!'
    },
    {
      title: 'Section 3',
      content:
        'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?'
    }
  ];
  it('render without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(<Accordion />, div);
    ReactDom.unmountComponentAtNode(div);
  });
  it('renders the UI without props', () => {
    const wrapper = shallow(<Accordion />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('render the UI with props', () => {
    const wrapper = shallow(<Accordion sections={sections} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('render the UI with props on click second button', () => {
    const wrapper = shallow(<Accordion sections={sections} />);
    wrapper.find('button').at(1).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('render the Ui with props clicking the last one after the second', () => {
    const wrapper = shallow(<Accordion sections={sections} />);
    wrapper.find('button').at(1).simulate('click');
    wrapper.find('button').at(2).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
