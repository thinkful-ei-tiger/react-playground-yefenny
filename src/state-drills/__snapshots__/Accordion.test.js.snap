// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component render the UI with props 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion component render the UI with props on click second button 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion component render the Ui with props clicking the last one after the second 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Accordion component renders the UI without props 1`] = `<ul />`;
